internal/service/advancedclustertpf/resource_compatiblity.go

```diff
@@ -16,6 +16,10 @@ func overrideKnowTPFIssueFields(modelIn, modelOut *TFModel) {
 	if beforeVersion != nil && !modelIn.MongoDBMajorVersion.Equal(modelOut.MongoDBMajorVersion) {
 		modelOut.MongoDBMajorVersion = types.StringPointerValue(beforeVersion)
 	}
+	retainBackups := conversion.NilForUnknown(modelIn.RetainBackupsEnabled, modelIn.RetainBackupsEnabled.ValueBoolPointer())
+	if retainBackups != nil && !modelIn.RetainBackupsEnabled.Equal(modelOut.RetainBackupsEnabled) {
+		modelOut.RetainBackupsEnabled = types.BoolPointerValue(retainBackups)
```

Comment:
Could we have the parent function have a more specific name? Potentially `overrideAttributesWithPrevStateValue`.

Reply:
makes sense, changed in a4dd9e0f06ca3db6edcfa356322556f741d6ac95